src/coreclr/jit/lsra.h

```diff
@@ -733,6 +733,7 @@ class LinearScan : public LinearScanInterface
     unsigned int currentSpill[TYP_COUNT];
     bool         needFloatTmpForFPCall;
     bool         needDoubleTmpForFPCall;
+    bool         needFloatingRegisters;
```

Comment:
A safe change would be to have `needFloatingRegisters |= (theRegisterType != IntRegisterType);` in `newInterval()` method but want to see if my current changes achieve the purpose.